src/org/zalando/stups/mint/worker/job/sync_app.clj

```diff
@@ -20,7 +20,7 @@
     (if-not (> s3_errors
                max-s3-errors)
             (let [app        (storage/get-app storage-url id tokens)
-                  unwritable (doall (remove #(writable? % id) (:s3_buckets app)))]
+                  unwritable (doall (remove #(writable? % id configuration tokens) (:s3_buckets app)))]
```

Comment:
I guess the same changes is also necessary for the invocations of `save-user` and `save-client`?